File: php/roman-numerals/roman-numerals.php
<?php

function translateDigit(&$num, $digit) {
    static $translate = [
        1    => ['', 'I', 'II', 'III', 'IV', 'V', 'VI', 'VII', 'VIII', 'IX', 'X'],
        10   => ['', 'X', 'XX', 'XXX', 'XL', 'L', 'LX', 'LXX', 'LXXX', 'XC', 'C'],
        100  => ['', 'C', 'CC', 'CCC', 'CD', 'D', 'DC', 'DCC', 'DCCC', 'CM', 'M'],
        1000 => ['', 'M', 'MM', 'MMM']
    ];

    $count  = floor($num / $digit);
    $num   -= $count * $digit;
    
    return $translate[$digit][$count];
}

function toRoman($num) {
    if (!is_numeric($num) || $num < 1 || $num > 3000) {
        throw new \InvalidArgumentException('Only numbers between 1 and 3000 can be converted.');
    }

    $num = (int)$num;
    
    $roman  = '';
    $roman .= translateDigit($num, 1000);
    $roman .= translateDigit($num, 100);
    $roman .= translateDigit($num, 10);
    $roman .= translateDigit($num, 1);
    
    return $roman;
}

File: fsharp/roman-numerals/RomanNumerals.fs
module RomanNumerals

let roman arabicNumeral =
    let singles arabicNumeral =
        match (arabicNumeral % 10) with
        | 1 -> "I"
        | 2 -> "II"
        | 3 -> "III"
        | 4 -> "IV"
        | 5 -> "V"
        | 6 -> "VI"
        | 7 -> "VII"
        | 8 -> "VIII"
        | 9 -> "IX"
        | _ -> ""

    let tens arabicNumeral =
        match (arabicNumeral % 100) / 10 with
        | 1 -> "X"
        | 2 -> "XX"
        | 3 -> "XXX"
        | 4 -> "XL"
        | 5 -> "L"
        | 6 -> "LX"
        | 7 -> "LXX"
        | 8 -> "LXXX"
        | 9 -> "XC"
        | _ -> ""

    let hundreds arabicNumeral =
        match (arabicNumeral % 1000) / 100 with
        | 1 -> "C"
        | 2 -> "CC"
        | 3 -> "CCC"
        | 4 -> "CD"
        | 5 -> "D"
        | 6 -> "CD"
        | 7 -> "CDD"
        | 8 -> "CDDD"
        | 9 -> "CM"
        | _ -> ""

    let thousands arabicNumeral =
        match arabicNumeral / 1000 with
        | 1 -> "M"
        | 2 -> "MM"
        | 3 -> "MMM"
        | _ -> ""

    System.String.Concat [thousands arabicNumeral; hundreds arabicNumeral; tens arabicNumeral; singles arabicNumeral]